build.gradle.kts

```
@@ -47,4 +60,76 @@ val integrationTestImplementation: Configuration by configurations.getting {
 }
 val integrationTestRuntimeOnly: Configuration by configurations.getting {
     extendsFrom(configurations.testRuntimeOnly.get())
-}
\ No newline at end of file
+}
+val containerName = findProperty("containerName") ?: "hivemq-k8s-test"
+val containerTag = findProperty("containerTag") ?: "snapshot"
+
+val producerK8sDockerImage: Configuration by configurations.creating {
+    isCanBeConsumed = false
+    isCanBeResolved = true
+    attributes {
+        attribute(Category.CATEGORY_ATTRIBUTE, objects.named("k8s-docker-image"))
+    }
+    extendsFrom(hivemq)
+}
+val producerDnsInitWaitDockerImage: Configuration by configurations.creating {
+    isCanBeConsumed = false
+    isCanBeResolved = true
+    attributes {
+        attribute(Category.CATEGORY_ATTRIBUTE, objects.named("dns-init-wait-docker-image"))
+    }
+    extendsFrom(operator)
+}
+val producerOperatorDockerImage: Configuration by configurations.creating {
+    isCanBeConsumed = false
+    isCanBeResolved = true
+    attributes {
+        attribute(Category.CATEGORY_ATTRIBUTE, objects.named("operator-docker-image"))
+    }
+    extendsFrom(operator)
+}
+
+
+val createRootlessK8sImageContext by tasks.registering(Sync::class) {
+    group = "container"
+    description = "Prepare hivemq rootless k8s image context"
+    into(layout.buildDirectory.dir("container/context"))
+    from("container")
+}
+
+val buildRootlessK8sImage by tasks.registering(Exec::class) {
+    group = "container"
+    description = "Build hivemq rootless k8s image"
+    inputs.property("dockerImageName", containerName)
+    inputs.dir(createRootlessK8sImageContext.map { it.destinationDir })
+    workingDir(createRootlessK8sImageContext.map { it.destinationDir })
+    commandLine("docker", "build", "-f", "broker.dockerfile", "-t", "${containerName}-rootless:${containerTag}", ".")
+}
+
+val saveRootlessK8sImage by tasks.registering(Exec::class) {
+    group = "container"
+    description = "Save hivemq rootless k8s image"
+    dependsOn(buildRootlessK8sImage)
+    workingDir(layout.buildDirectory.dir("containers"))
+    commandLine("docker", "save", "-o", "${containerName}-rootless.tar", "${containerName}-rootless:${containerTag}")
+}
+val buildContainersFiles by tasks.registering(Copy::class) {
+    group = "container"
+    from(producerK8sDockerImage)
+    from(producerDnsInitWaitDockerImage)
+    dependsOn(gradle.includedBuild("hivemq-operator").task(":jibBuildTar"))
+    from(
+        producerK8sDockerImage.singleFile,
+        producerDnsInitWaitDockerImage.singleFile,
+        producerOperatorDockerImage.singleFile
+    )
```

```suggestion
    from(producerK8sDockerImage) { rename { "hivemq-k8s-image.tar" } }
    from(producerDnsInitWaitDockerImage) { rename { "hivemq-init-dns-image.tar" } }
    from(producerOperatorDockerImage) { rename { "hivemq-operator.tar" } }
```
Please rename the files here if you require a specific file name instead of renaming the artifact files itself.
Also the dependency on jibBuildTar is automatically added if it is correctly configured in the hivemq-operator repository. I added a comment there.